test/interpreter/separate-cache.spec.ts

```diff
@@ -0,0 +1,17 @@
+import {HyperFormula} from '../../src'
+import {adr} from '../testUtils'
+
+describe('numeric aggreagtion functions', () => {
+  it('should use separate caches', () => {
+    const engine = HyperFormula.buildFromArray([
+      [1, 2, 5, 10, 20],
+      ['=MIN(A1:E1)', '=MAX(A1:E1)', '=SUM(A1:E1)', '=SUMSQ(A1:E1)', '=AVERAGE(A1:E1)'],
+      ['=MIN(A1:E1)', '=MAX(A1:E1)', '=SUM(A1:E1)', '=SUMSQ(A1:E1)', '=AVERAGE(A1:E1)'],
+    ])
+    expect(engine.getCellValue(adr('A3'))).toEqual(1)
+    expect(engine.getCellValue(adr('B3'))).toEqual(20)
+    expect(engine.getCellValue(adr('C3'))).toEqual(38)
+    expect(engine.getCellValue(adr('D3'))).toEqual(530)
+    expect(engine.getCellValue(adr('E3'))).toEqual(7.6)
```

Comment:
Can similar problem occur when functions are the same but ranges differ? Looks like one row is unused, so I have a suggestion:

```suggestion
      [1, 2, 5, 10, 20],
      ['=MIN(B1:E1)', '=MAX(A1:D1)', '=SUM(B1:E1)', '=SUMSQ(A1:D1)', '=AVERAGE(A1:D1)'],
      ['=MIN(A1:E1)', '=MAX(A1:E1)', '=SUM(A1:E1)', '=SUMSQ(A1:E1)', '=AVERAGE(A1:E1)'],
    ])
    
    expect(engine.getCellValue(adr('A2'))).toEqual(2)
    expect(engine.getCellValue(adr('B2'))).toEqual(10)
    expect(engine.getCellValue(adr('C2'))).toEqual(37)
    expect(engine.getCellValue(adr('D2'))).toEqual( ?? )
    expect(engine.getCellValue(adr('E2'))).toEqual( ?? )
    
    expect(engine.getCellValue(adr('A3'))).toEqual(1)
    expect(engine.getCellValue(adr('B3'))).toEqual(20)
    expect(engine.getCellValue(adr('C3'))).toEqual(38)
    expect(engine.getCellValue(adr('D3'))).toEqual(530)
    expect(engine.getCellValue(adr('E3'))).toEqual(7.6)
```

Reply:
The reason why I copied the row is to have it so each function is called once on exactly one particular range, before we try to call them second time, (so we check possible cache pollution for every combination).

Reply:
> Can similar problem occur when functions are the same but ranges differ? Looks like one row is unused, so I have a suggestion:

If it would, it would also occur with identical ranges.

Reply:
ok then 👍